frontend/javascripts/oxalis/view/right-border-tabs/segments_tab/segment_statistics_modal.tsx

```diff
@@ -87,65 +111,84 @@ export function SegmentStatisticsModal({
-            dataSetScale,
-            layersFinestResolution,
-            currentSegmentSizeInVx,
-          );
-          const currentGroupId = getGroupIdForSegment(currentSegment);
-          const segmentStateObject = {
-            key: currentSegment.id,
-            segmentId: currentSegment.id,
-            segmentName:
-              currentSegment.name == null ? `Segment ${currentSegment.id}` : currentSegment.name,
-            groupId: currentGroupId,
-            groupName: getGroupNameForId(currentGroupId),
-            volumeInVoxel: currentSegmentSizeInVx,
-            volumeInNm3,
-            formattedSize: formatNumberToVolume(volumeInNm3),
-            boundingBoxTopLeft: boundingBoxInMag1.topLeft,
-            boundingBoxTopLeftAsString: `(${boundingBoxInMag1.topLeft.join(", ")})`,
-            boundingBoxPosition: [
-              boundingBoxInMag1.width,
-              boundingBoxInMag1.height,
-              boundingBoxInMag1.depth,
-            ] as Vector3,
-            boundingBoxPositionAsString: `(${boundingBoxInMag1.width}, ${boundingBoxInMag1.height}, ${boundingBoxInMag1.depth})`,
-          };
-          statisticsObjects.push(segmentStateObject);
-        }
-        return statisticsObjects;
-      });
+      ]).then(
+        (response) => {
+          const segmentSizes = response[0];
+          const boundingBoxes = response[1];
+          const statisticsObjects = [];
+          const additionalCoordStringForCsv =
+            getAdditionalCoordinatesAsString(additionalCoordinates);
+          for (let i = 0; i < segments.length; i++) {
+            // segments in request and their statistics in the response are in the same order
+            const currentSegment = segments[i];
+            const currentBoundingBox = boundingBoxes[i];
+            const boundingBoxInMag1 = getBoundingBoxInMag1(
+              currentBoundingBox,
+              layersFinestResolution,
+            );
+            const currentSegmentSizeInVx = segmentSizes[i];
+            const volumeInNm3 = voxelToNm3(
+              dataSetScale,
+              layersFinestResolution,
+              currentSegmentSizeInVx,
+            );
+            const currentGroupId = getGroupIdForSegment(currentSegment);
+            const segmentStateObject: SegmentInfo = {
+              key: currentSegment.id,
+              additionalCoordinates: additionalCoordStringForCsv,
+              segmentId: currentSegment.id,
+              segmentName:
+                currentSegment.name == null ? `Segment ${currentSegment.id}` : currentSegment.name,
+              groupId: currentGroupId,
+              groupName: getGroupNameForId(currentGroupId),
+              volumeInVoxel: currentSegmentSizeInVx,
+              volumeInNm3,
+              formattedSize: formatNumberToVolume(volumeInNm3),
+              boundingBoxTopLeft: boundingBoxInMag1.topLeft,
+              boundingBoxTopLeftAsString: `(${boundingBoxInMag1.topLeft.join(", ")})`,
```

Comment:
`boundingBoxTopLeftAsString` and `boundingBoxPositionAsString` are written here, but they are never used, are they? at least, I don't see it currently.

Reply:
Both are used in the columns of the table in the modal, but referenced via `dataIndex`